### PR TITLE
MYREC-3314: Choose GET implementation

### PR DESCRIPTION
**Pull request by penevAt**
Added a UnirecsParameterArgumentResolver which gets applied to parameters annotated with @UnirecsParameterObject.

### CODE REVIEW

File: src/main/java/uk/co/bbc/unirecs/chassis/config/AppConfig.java [hunk 1]
```diff
@@ -3,16 +3,42 @@
...
+        argumentResolvers.add(new UnirecsParameterArgumentResolver(objectMapper()));
+    }
+
+    @Bean
+    public ConfigurableServletWebServerFactory webServerFactory() {
+        final TomcatServletWebServerFactory factory = new TomcatServletWebServerFactory();
+        factory.addConnectorCustomizers(new TomcatConnectorCustomizer() {
+            @Override
+            public void customize(final Connector connector) {
+                connector.setProperty("relaxedQueryChars", "[]");
```
Comment (SamGordonSmith) [inline src/main/java/uk/co/bbc/unirecs/chassis/config/AppConfig.java]: What are we using this for?

Comment (SamGordonSmith) [reply src/main/java/uk/co/bbc/unirecs/chassis/config/AppConfig.java]: Ah I see! What part actually resolves the query parameters to the java object for the request? Is that also done in the UnirecsParameterArgumentResolver?

Comment (penevAt) [reply src/main/java/uk/co/bbc/unirecs/chassis/config/AppConfig.java]: Yeah, it's done in the UnirecsParameterArgumentResolver.
The resolver checks if the request contains a "payload" param, if it does then it base64 decodes it and returns the ChassisRecommendationsRequest.

Comment (SamGordonSmith) [reply src/main/java/uk/co/bbc/unirecs/chassis/config/AppConfig.java]: clever stuff!

**Pull request merged**